src/controllers/task.list.controller.ts

```diff
@@ -20,9 +20,9 @@ export const get = async (req: Request, res: Response) => {
 };
 
 const calculateFilingDate = (recordDate: string, companyProfile: CompanyProfile): string => {
-  const nextMadeUpToDate = companyProfile.confirmationStatement.nextMadeUpTo;
-  if (nextMadeUpToDate && isInFuture(nextMadeUpToDate)) {
+  const nextMadeUpToDate = companyProfile.confirmationStatement!.nextMadeUpTo;
```

Comment:
i'd still have the checks for undefined in place, just in case.
You can use :
`const nextMadeUpToDate = companyProfile?.confirmationStatement?.nextMadeUpTo;`
which will return undefined if any of the question marked items are undefined and won't throw any errors.
This should be safe as isInFuture is safe handling undefined, it will return false if given an undefined (from looking at the unit test)

Reply:
Im not sure as if i remove the ! after confirmationstatement i get the expected
> Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
  Type 'undefined' is not assignable to type 'string'.

for the isInFuture method, it doesnt seem to be actually able to handle an undefined

Reply:
Also in the case that somehow a company does have an undefined confirmation statement, it will direct them to service unavailable anyway which i believe is the correct procedure